zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationPreconditionChecker.java

```diff
@@ -464,27 +465,34 @@ public static void requireUnchangedFlowScope(
 
   /**
    * Checks whether the given source process definition contains a boundary event. Throws an
-   * exception if the source process definition contains a boundary event.
+   * exception if the source process definition contains a boundary event that is not allowed.
    *
    * @param sourceProcessDefinition source process definition to do the check
    * @param elementInstanceRecord element instance to be logged
+   * @param allowedEventTypes allowed event types for the boundary event
    */
   public static void requireNoBoundaryEventInSource(
       final DeployedProcess sourceProcessDefinition,
-      final ProcessInstanceRecord elementInstanceRecord) {
-    final boolean hasBoundaryEventInSource =
-        !sourceProcessDefinition
+      final ProcessInstanceRecord elementInstanceRecord,
+      final EnumSet<BpmnEventType> allowedEventTypes) {
+    final List<String> disallowedBoundaryEventsInSource =
```

Comment:
💭 I found this naming too long. Can we replace it with something simpler? E.g. `restrictedBoundaryEvents`.